### PR TITLE
Catch connection refused during asd set

### PR DESCRIPTION
**Pull request by JeffreyDevloo**
#212

### CODE REVIEW

File: ovs/extensions/healthcheck/alba/alba_health_check.py [hunk 1]
```diff
@@ -376,10 +376,13 @@ def check_backend_asds(logger, disks, backend_name):
...
-                                        named_params={'host': ip_address, 'port': str(disk.get('port')),
-                                                      'long-id': disk.get('asd_id')},
-                                        extra_params=[key, value])
+                            try:
+                                AlbaCLI.run(command="asd-set",
+                                            named_params={'host': ip_address, 'port': str(disk.get('port')),
+                                                          'long-id': disk.get('asd_id')},
+                                            extra_params=[key, value])
+                            except RuntimeError:
+                                raise ConnectionFailedException('Connection failed to disk')
```
Comment (domsj) [inline ovs/extensions/healthcheck/alba/alba_health_check.py]: not all runtime errors indicate connection to the disk has failed

Comment (pploegaert) [inline ovs/extensions/healthcheck/alba/alba_health_check.py]: does alba gives us a useful exception?

Comment (JeffreyDevloo) [inline ovs/extensions/healthcheck/alba/alba_health_check.py]: The only way to differentiate between errors that our wrapper returns is by looking at the supplied message by alba.
For that we would need a list of possible messages that alba returns. Could you provide us that?

Comment (domsj) [inline ovs/extensions/healthcheck/alba/alba_health_check.py]: as discussed, you don't need a full list of errors.
I could give you one that's exhaustive today, but that's no guarantee for the future. So you just want to use our message in the exception you throw.

File: ovs/extensions/healthcheck/alba/alba_health_check.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def _fetch_available_backends(logger):
                                     asd['port'] = Configuration.get('/ovs/alba/asds/{0}/config|port'.format(asd_id))
                                     disks.append(asd)
                                 except Exception as ex:
-                                    raise ConnectionFailedException(ex)
+                                    raise ConnectionFailedException(str(ex))
```
Comment (domsj) [inline ovs/extensions/healthcheck/alba/alba_health_check.py]: wrong exception type?

File: ovs/extensions/healthcheck/alba/alba_health_check.py [hunk 3]
```diff
@@ -96,9 +96,14 @@ def _fetch_available_backends(logger):
...
+                                except NotFoundException as ex:
+                                    logger.failure("Could not find {0} in Arakoon. Got {1}"
+                                                   .format('/ovs/alba/asds/{0}/config|port'.format(asd_id), str(ex)))
+                                    raise
                                 except Exception as ex:
-                                    raise ConnectionFailedException(ex)
-                # create result
+                                    logger.failure("Could not connect to the Arakoon.")
+                                    raise ConnectionFailedException(str(ex))
+               # create result
```
Comment (pploegaert) [inline ovs/extensions/healthcheck/alba/alba_health_check.py]: indent incorrect for comment ...

**Pull request merged**